Develop Anari

nicolemarsaglia opened the pull request.

nicolemarsaglia committed:
add fields from other domains

nicolemarsaglia committed:
conflict

nicolemarsaglia committed:
push before switching

nicolemarsaglia committed:
let's put this here for now

nicolemarsaglia committed:
blt_mpi_deps

nicolemarsaglia committed:
add logic for if vtkm has mpi on

nicolemarsaglia committed:
Update SetupMFEM.cmake

depts->deps

nicolemarsaglia committed:
try it this way?

nicolemarsaglia committed:
Merge branch 'task/2025_03_warpx_unit_test' of https://github.com/Alpine-DAV/ascent into task/2025_03_warpx_unit_test

nicolemarsaglia committed:
idk this is weird

nicolemarsaglia committed:
lets see this pass the ci then I'll fix the cmakelists

nicolemarsaglia committed:
Merge branch 'task/2025_03_warpx_unit_test' of https://github.com/Alpine-DAV/ascent into task/2025_03_warpx_unit_test

nicolemarsaglia committed:
lets try this cmake flag again for if mpi is not found

nicolemarsaglia committed:
Merge branch 'task/2025_03_warpx_unit_test' of https://github.com/Alpine-DAV/ascent into task/2025_03_warpx_unit_test

nicolemarsaglia committed:
t- => t_

nicolemarsaglia committed:
Merge branch 'develop' into task/2025_03_warpx_unit_test

nicolemarsaglia committed:
Merge branch 'develop' into task/2025_03_warpx_unit_test

nicolemarsaglia committed:
remove comment

nicolemarsaglia committed:
Develop anari (#1538)

Initial ANARI development via VTKm/Viskores

nicolemarsaglia committed:
remove cerr statments

nicolemarsaglia committed:
Merge branch 'develop' into task/2025_05_develop_anari

nicolemarsaglia committed:
revert this change

nicolemarsaglia committed:
remove print statements

nicolemarsaglia committed:
add anari as rendering option

nicolemarsaglia committed:
remove encoder

cyrush commented on scripts/build_ascent/build_ascent.sh:
```diff
@@ -1056,6 +1103,11 @@ if ${build_catalyst}; then
 fi
 
+if ${build_anari}; then
+    echo 'set(ENABLE_ANARI ON CACHE BOOL "")' >> ${root_dir}/ascent-config.cmake
```
We should only key off of `ANARI_DIR`, we don't need an extra switch to enable to disable anari.

cyrush commented on src/CMakeLists.txt:
```diff
@@ -43,13 +43,17 @@ option(ENABLE_HIDDEN_VISIBILITY "Build with hidden visibility for private symbol
 option(ENABLE_CUDA_DEBUG_CPU_ONLY "Enable CUDA CPU debugging" OFF)
 
+# support ANARI rendering backend
+option(ENABLE_ANARI "Build ANARI Support" OFF)
```
we dont need addtional option, just key off ANARI_DIR

cyrush commented on src/config/ascent_setup_deps.cmake:
```diff
@@ -188,6 +188,35 @@ if(KOKKOS_DIR)
+    set(ANARI_DIR ${ASCENT_ANARI_DIR})
+endif()
+
+if(EXISTS ${ANARI_DIR}/lib64/cmake/Kokkos/)
```
is kokkos a typo here?

nicolemarsaglia replied:
bingo! copy/paste typo

cyrush commented on src/libs/ascent/runtimes/ascent_main_runtime.cpp:
```diff
@@ -1101,10 +1112,6 @@ AscentRuntime::ConvertExtractToFlow(const conduit::Node &extract,
-  else if(extract_type == "steering")
-  {
-     filter_name = "steering";
-  }
```
why is this guy getting removed?

nicolemarsaglia replied:
Weird. I don't know. Probably messed up during a merge or something. 

Thanks for catching

nicolemarsaglia committed:
Merge branch 'develop' into task/2025_05_develop_anari

nicolemarsaglia committed:
kokkos typo

nicolemarsaglia committed:
remove anari as IsVolume in Scene.cpp

remove this check for now while we test and decide how to design things. 

anari vs. anari_vol & anari_pseudo & anari_points(?) idk all their renderers

nicolemarsaglia committed:
add triangle renderer

nicolemarsaglia committed:
start to get color map working

nicolemarsaglia committed:
update

nicolemarsaglia committed:
combine anari to a single renderer

nicolemarsaglia committed:
move device initializion from constructor to execture so we only do it once.

nicolemarsaglia committed:
add anari mapper triangles patch

nicolemarsaglia committed:
add ptc to build script

nicolemarsaglia committed:
add build scripts

nicolemarsaglia committed:
add tests

Commit:
add stuff

nicolemarsaglia committed:
update against develop

nicolemarsaglia committed:
update anari ptc barney versions

nicolemarsaglia committed:
Merge branch 'develop' into task/2025_05_develop_anari

nicolemarsaglia committed:
Merge branch 'develop' into task/2025_05_develop_anari

nicolemarsaglia committed:
remove odl patch
